github/resource_github_repository_ruleset.go

```diff
@@ -252,6 +252,51 @@ func resourceGithubRepositoryRuleset() *schema.Resource {
 								},
 							},
 						},
+						"merge_queue": {
```

Comment:
two things that are different from the version I current have locally:
1. all the elements under `Elem` can actually be optional, when you enable a merge queue without changing any settings, github set default values for each configurations
2. although not explicitly included in the API documentation, all of the `schema.TypeInt` types have minimum and maximum value constraints, you can know the values if you play with the UI and try to set the value to a very high number. 

happy to discuss further or provide my version of the code if necessary.

Reply:
1. Ah interesting, I was basing it off the rest API docs that listed every attribute as required.
2. Will have a play and find those, thanks.
3. Will work out where to add those.

Reply:
@zepeng811 One thing I am not quite clear on is preference of using `ValidateFunc` vs `ValidateDiagFunc`.  If I understand correctly, `ValidateDiagFunc` gives better contextual errors to the user?  I see both being used in this provider though.

Reply:
I'm not a contributor for this repo, but from what I see `ValidateDiagFunc` is used way more frequently than `ValidateFunc`:
* ValidateDiagFunc: https://github.com/search?q=repo%3Aintegrations%2Fterraform-provider-github+%22ValidateDiagFunc%22&type=code&p=3
* ValidateFunc: https://github.com/search?q=repo%3Aintegrations%2Fterraform-provider-github+%22ValidateFunc%22&type=code

so I assume `ValidateFunc` might be a deprecated function? someone else more familiar with the repo can confirm.

Reply:
That's right, `ValidateDiagFunc` is the new thing from Hashicorp that allows collecting diagnostic information with validations. We'd like to stick with that over `ValidateFunc`.